src/batched/sparse/impl/KokkosBatched_GMRES_TeamVector_Impl.hpp

```diff
@@ -181,9 +185,12 @@ struct TeamVectorGMRES {
             Kokkos::TeamVectorRange(member, 0, numMatrices),
```

Comment:
We need an extra `member.team_barrier();` before the loop: 
```
        Kokkos::parallel_for(
            Kokkos:: TeamVectorRange(member, 0, numMatrices),
            [&](const OrdinalType& ii) { tmp(ii) = -tmp(ii); });
```
Otherwise, the copy before can copy some `tmp` with changed sign.